includes/transformer/class-post.php

```diff
@@ -208,7 +209,67 @@ protected function get_image() {
 		 */
 		$thumbnail = apply_filters(
 			'activitypub_get_image',
-			self::get_wordpress_attachment( $id, $image_size ),
+			$this->get_wordpress_attachment( $id, $image_size ),
+			$id,
+			$image_size
+		);
+
+		if ( ! $thumbnail ) {
+			return null;
+		}
+
+		$mime_type = \get_post_mime_type( $id );
+
+		$image = array(
+			'type'      => 'Image',
+			'url'       => \esc_url( $thumbnail[0] ),
+			'mediaType' => \esc_attr( $mime_type ),
+		);
+
+		$alt = \get_post_meta( $id, '_wp_attachment_image_alt', true );
+		if ( $alt ) {
+			$image['name'] = \wp_strip_all_tags( \html_entity_decode( $alt ) );
+		}
+
+		return $image;
+	}
+
+	/**
+	 * Returns an Icon, based on the Featured Image with a fallback to the site-icon.
+	 *
+	 * @return array|null The Icon or null if no icon is available.
+	 */
+	protected function get_icon() {
+		$post_id = $this->wp_object->ID;
+		$id      = null;
```

Comment:
`$id` gets never used, it's either the post thumbnail or the the option later on.